### PR TITLE
fix canonical revision label for workload groups

### PR DESCRIPTION
**Pull request by douglas-reid**
An incorrect label was used in the `workload` command, leading to incorrectly labeled `WorkloadGroup`s. This PR corrects the issue, first identified in #34395.

- [ X ] Policies and Telemetry

### CODE REVIEW

**Review by bianpengyuan — APPROVED** [review]
Do we have any integration test for this? Also probably worth a release notes.

File: istioctl/cmd/workload.go [hunk 1]
```diff
@@ -530,7 +530,7 @@ func createMeshConfig(kubeClient kube.ExtendedClient, wg *clientv1alpha3.Workloa
 	}
 	md["ISTIO_META_WORKLOAD_NAME"] = wg.Name
 	labels["service.istio.io/canonical-name"] = md["CANONICAL_SERVICE"]
-	labels["service.istio.io/canonical-version"] = md["CANONICAL_REVISION"]
+	labels["service.istio.io/canonical-revision"] = md["CANONICAL_REVISION"]
```
Comment (howardjohn) [inline istioctl/cmd/workload.go:533]: do we have this in istio.io/api/labels package? would avoid this

**Review by howardjohn — APPROVED** [review]

**Comment by douglas-reid** [comment]
> Do we have any integration test for this? Also probably worth a release notes.

I don't believe we have any integration tests for the `workload` command (or generated telemetry from the impacted workloads). There are the golden file-based unit tests updated in this PR, but that's all I'm aware of.

Release notes have been added.

**Review by stevenctl — APPROVED** [review]

**Comment by stevenctl** [comment]
The VMs in integration tests are set up using this code, so it should be possible to test the generated telemetry.

**Comment by douglas-reid** [comment]
> The VMs in integration tests are set up using this code, so it should be possible to test the generated telemetry.

Do you have a quick pointer?  The telemetry tests on VMs are limited, and seem to manually label instances ([example](https://github.com/istio/istio/blob/master/tests/integration/telemetry/stackdriver/vm/main_test.go#L166)).

**Comment by stevenctl** [comment]
The things in `vmEnv` just get appended to the output from `workload` command:

https://github.com/istio/istio/blob/4ccc36879f85faba3e78ed0b96c8230559be6d63/pkg/test/framework/components/echo/kube/deployment.go#L983

You can (probably) remove a good amount of that customization and cover what we generate in the command.

**Comment by douglas-reid** [comment]
> You can (probably) remove a good amount of that customization and cover what we generate in the command.

Sounds good. Filed #35319 to track follow-up to remove customization.

**Review by therealmitchconnors — APPROVED** [review]

**Pull request merged**